File: frontend/src/types/recommendation.ts
import type { Facility } from './facility';

export interface FacilityRecommendation {
  id: number;
  name: string;
  category: string;
  status: Facility['status'];
  rating: number | null;
  distance_m: number;
  estimated_time_s: number;
  travel_source: 'network' | 'straight_line_estimate';
  predicted_usage: number;
  crowd_level: string;
  prediction_source: string;
  recommendation_score: number;
  rank_position: number;
}

export interface RecommendationResponse {
  recommended_facility: FacilityRecommendation | null;
  ranked_facilities: FacilityRecommendation[];
  explanation: string | null;
  message: string | null;
}

File: backend/app/schemas/recommendation.py
"""Response schemas for the recommendation endpoint."""

from pydantic import BaseModel


class FacilityRecommendation(BaseModel):
    """A single scored facility in the recommendation response."""

    id: int
    name: str
    category: str
    status: str
    rating: float | None
    distance_m: float
    estimated_time_s: float
    travel_source: str
    predicted_usage: float
    crowd_level: str
    prediction_source: str
    recommendation_score: float
    rank_position: int


class RecommendationResponse(BaseModel):
    """Top-level response for GET /api/v1/recommendations."""

    recommended_facility: FacilityRecommendation | None
    ranked_facilities: list[FacilityRecommendation]
    explanation: str | None
    message: str | None
